### PR TITLE
Round 2 bugs

### PR DESCRIPTION
**Pull request by znewsham**
### `observe.ts` — `addedAt` index is off-by-one (might matter)
**Symptom:** the `atIndex` argument passed to `addedAt` is always `(actual_index + 1)`. `cache.size()` and `cache.indexOf(before)` are computed *after* `cache.addedBefore`, so both reflect the post-insert state.

### `subscriber.ts:124` — sort-as-array silently breaks projection (unlikely to matter)
The `sort` type allows `[string, 1|-1][]`. With an array sort, `Object.entries(sort)` yields `["0", ["field", 1]]`, so the sort projection becomes `{ "0": 1, "1": 1 }` instead of `{ field: 1 }`. `combinedProjection` and `cursor.project(...)` degrade silently.

### `subscriber.ts:373` — `this.#matcher._path` crashes when no filter (will matter - sort + no filter + changes would break)
The matcher is only constructed when `cursor.cursorDescription.filter` is set (line 143). A LIMIT_SORT cursor with sort+limit but no filter throws `TypeError: Cannot read properties of undefined (reading '_path')` on the first requery. `_path` is also private API of Matcher — brittle.

### `observe.ts` — falsy treatment of `before === 0` / `""` (unlikely to matter - these fall into the same "thanks typescript" category)
`before ? cache.indexOf(before) : cache.size()` and
`before && transform(cache.get(before))` collapse legitimate IDs of `0` or
`""` to "no before". `Stringable` allows `number | boolean | string | …`,
so this is reachable.

### `utils.ts:34` — Date `_id` falls into "EMPTY" branch (unlikely to matter)
`Object.keys(new Date())` is `[]`, so a Date `_id` is dropped from
`extractIdsFromSelector`. Subscribers with Date IDs silently regress to
DEFAULT-strategy semantics.

### `subscriber.ts` — `#docs` is declared but never assigned (dead code cleanup)
`#docs: OrderedDict<...> | undefined` — never assigned anywhere. Every `if (this.#docs)` branch (in `#has`, `#get`, `#size`) is dead. Misleading to readers.

### `publish.ts` — mutator parsing crashes on pipeline
Assumes `mutator` is `{ $set: {...}, $inc: {...} }`. Other valid Mongo inputs break:
- Pipeline-form update (`[ { $set: {...} } ]`) — `Object.values(arr)` works
  but `key.split` would receive numeric-string keys.

---

> [!NOTE]
> **Medium Risk**
> Touches core observe/Redis reactivity paths (ordered indexing, subscriber projections, and published UPDATE metadata), so regressions could silently drop or misorder live updates despite added tests.
> 
> **Overview**
> Fixes several correctness issues in ordered observation and Redis-based live updates.
> 
> In `observe.ts`, `addedAt` now reports the pre-insert index (and treats `before` values like `0`/`""` as valid), eliminating off-by-one and falsy-ID bugs; new tests assert correct indices.
> 
> In Redis publish/subscribe, UPDATE messages now compute top-level changed fields via `topLevelFieldsFromMutator` (including pipeline-form updates) and omit `FIELDS` when the impacted field set can’t be determined; `RedisUpdate` is updated to make `FIELDS` optional. The subscriber also now supports array-form `sort`, avoids requery crashes when there is no matcher by using `completeProjection`, and removes unused `#docs` branches. Additional regression tests cover pipeline mutators, Date `_id` channel extraction, and sort/requery edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 748e4371bede9bd839a9430ef2230a80cb2d0c2b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/redis/publish.ts [hunk 1]
```diff
@@ -58,10 +58,62 @@ export async function handleRemove(
...
+              if (typeof path === "string") fields.add(path.split(".")[0]);
+            }
+          }
+          continue;
+        }
+        if (op === "$set" || op === "$addFields" || op === "$project") {
+          if (operand && typeof operand === "object" && !Array.isArray(operand)) {
+            for (const key of Object.keys(operand)) fields.add(key.split(".")[0]);
+          }
+          continue;
```
[the remark this answers is not in the document]
Comment (znewsham) [reply src/redis/publish.ts:89]: @cursor fix this
